File: tmp/preview/v3/add-tablet-h-layout.js
const fs = require('fs');
const path = require('path');
const file = path.join(__dirname, 'home-html-real-mockup-v4.html');
let text = fs.readFileSync(file, 'utf8');
const marker = '/* Tablet horizontal — mismo layout y proporciones que laptop */';
if (text.includes(marker)) {
  console.log('Tablet-h block already present');
  process.exit(0);
}
const startKey = 'body.proto-laptop .home-page-body{z-index:0}';
const endKey = 'body.proto-laptop .home-ad-bottom__dot{width:6px;height:6px}';
const start = text.indexOf(startKey);
const end = text.indexOf(endKey) + endKey.length;
const block = text.slice(start, end);
const lines = block.split(/\r?\n/);
const out = [marker];
let i = 0;
while (i < lines.length) {
  const line = lines[i];
  if (/body\.proto-(iphone|tablet)/.test(line)) {
    while (i < lines.length && !lines[i].includes('}')) i++;
    i++;
    continue;
  }
  if (line.includes('@media') && i + 1 < lines.length && lines[i + 1].includes('body.proto-laptop')) {
    while (i < lines.length) {
      out.push(lines[i].replace(/body\.proto-laptop/g, 'body.proto-tablet-h'));
      if (lines[i].trim() === '}') { i++; break; }
      i++;
    }
    continue;
  }
  if (line.includes('body.proto-laptop')) {
    while (i < lines.length) {
      out.push(lines[i].replace(/body\.proto-laptop/g, 'body.proto-tablet-h'));
      if (lines[i].includes('}')) { i++; break; }
      i++;
    }
    continue;
  }
  i++;
}
text = text.slice(0, end) + '\n' + out.join('\n') + '\n' + text.slice(end);
fs.writeFileSync(file, text, 'utf8');
console.log('Inserted', out.length, 'lines for tablet-h');

File: tmp/preview/v3/add-tablet-h-layout.py
#!/usr/bin/env python3
"""Duplicate body.proto-laptop layout rules for body.proto-tablet-h."""
from pathlib import Path
import re

path = Path(__file__).resolve().parent / 'home-html-real-mockup-v4.html'
text = path.read_text(encoding='utf-8')
marker = '/* Tablet horizontal — mismo layout y proporciones que laptop */'
if marker in text:
    print('Tablet-h block already present')
    raise SystemExit(0)

start = text.index('body.proto-laptop .home-page-body{z-index:0}')
end = text.index('body.proto-laptop .home-ad-bottom__dot{width:6px;height:6px}')
end += len('body.proto-laptop .home-ad-bottom__dot{width:6px;height:6px}')

block = text[start:end]
lines = block.splitlines()
out = [marker]
i = 0
while i < len(lines):
    line = lines[i]
    if re.search(r'body\.proto-(iphone|tablet)', line):
        while i < len(lines) and '}' not in lines[i]:
            i += 1
        i += 1
        continue
    if '@media' in line and i + 1 < len(lines) and 'body.proto-laptop' in lines[i + 1]:
        rule = []
        while i < len(lines):
            rule.append(lines[i].replace('body.proto-laptop', 'body.proto-tablet-h'))
            if lines[i].strip() == '}':
                i += 1
                break
            i += 1
        out.extend(rule)
        continue
    if 'body.proto-laptop' in line:
        rule = []
        while i < len(lines):
            rule.append(lines[i].replace('body.proto-laptop', 'body.proto-tablet-h'))
            if '}' in lines[i]:
                i += 1
                break
            i += 1
        out.extend(rule)
        continue
    i += 1

insert = '\n' + '\n'.join(out) + '\n'
text = text[:end] + insert + text[end:]
path.write_text(text, encoding='utf-8')
print(f'Inserted {len(out)} lines for tablet-h')
